openshift/dynamic/client.py

```diff
@@ -59,89 +61,22 @@ class DynamicClient(object):
         the kubernetes API
     """
 
-    def __init__(self, client):
+    def __init__(self, client, discoverer=None):
```

Comment:
Can you just default it to `LazyDiscoverer` instead of `None` here?

Reply:
The only reason I didn't do that is because the LazyDiscoverer class will get evaluated at load time and it makes ordering matter... So I'd need to move the Discoverer class definitions before the DynamicClient definition or move them to a separate file if we do this.

Reply:
That makes sense, could you just move the defaulting to the top of the function then, I think it makes it a little clearer that it's just fancy defaulting

Reply:
That makes sense, could you just move the defaulting to the top of the function then, I think it makes it a little clearer that it's just fancy defaulting